akka-actor-tests/src/test/scala/akka/io/DnsSpec.scala

```diff
@@ -0,0 +1,38 @@
+package akka.io
+
+import java.net.{ Inet4Address, Inet6Address, InetAddress }
+
+import scala.collection.immutable.Seq
+import org.scalatest.{ BeforeAndAfterAll, Matchers, WordSpec }
+
+class DnsSpec extends WordSpec with BeforeAndAfterAll with Matchers {
+
+  val ip4Address = InetAddress.getByAddress("localhost", Array[Byte](127, 0, 0, 1)) match {
+    case address: Inet4Address ⇒ address
+  }
+  val ipv6Address = InetAddress.getByAddress("localhost", Array[Byte](0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1)) match {
+    case address: Inet6Address ⇒ address
+  }
+
+  var temporaryValue: Option[String] = None
+
+  override def beforeAll() = {
+    temporaryValue = sys.props.get("java.net.preferIPv6Addresses")
+  }
+
+  override def afterAll() = temporaryValue match {
+    case Some(value) ⇒ sys.props.put("java.net.preferIPv6Addresses", value)
+    case _           ⇒ sys.props.remove("java.net.preferIPv6Addresses")
```

Comment:
This could be a problem if running tests in parallel. We could put this test into a multi-jvm test in the `akka-remote-tests` module so that it doesn't influence other tests.

@jpuerto If you are not familiar with multi-jvm tests we (Akka team) can fix this afterwards.